### PR TITLE
Add RAMP letter event and alerts to v2 API

### PR DESCRIPTION
**Pull request by cmgiven**
connects #4062 and #4063

### CODE REVIEW

File: app/models/appeal.rb [hunk 1]
```diff
@@ -237,6 +237,15 @@ def eligible_for_ramp?
     (status == "Advance" || status == "Remand") && !in_location?(:remand_returned_to_bva)
   end
 
+  def ramp_notice_date
+    @ramp_notice_date = RampElection.where(veteran_file_number: sanitized_vbms_id)
+                                    .minimum(:notice_date)
+  end
+
+  def ramp_due_date
+    ramp_notice_date + 60.days if ramp_notice_date
```
Comment (shanear) [inline app/models/appeal.rb]: ooh, we don't enforce this in the front end. Are you sure they have 60 days to opt in?

Comment (shanear) [reply app/models/appeal.rb]: if so, we should make that a constant in `RampElection`

```
RampElection::RECEIPT_REQUIRED_BEFORE = 60.days
```

Comment (cmgiven) [reply app/models/appeal.rb]: We're not enforcing that, no. But we want the alert to expire, so we're using the due date listed on the letter.

Comment (cmgiven) [reply app/models/appeal.rb]: Still prefer that be a constant associated with `RampElection`, given that it's a soft due date?

Comment (shanear) [reply app/models/appeal.rb]: ohhh, I'd make that clear in the name of the constant then. or a comment. Whatever.

File: app/models/appeal.rb [hunk 2]
```diff
@@ -237,6 +237,15 @@ def eligible_for_ramp?
     (status == "Advance" || status == "Remand") && !in_location?(:remand_returned_to_bva)
   end
 
+  def ramp_notice_date
+    @ramp_notice_date = RampElection.where(veteran_file_number: sanitized_vbms_id)
```
Comment (shanear) [inline app/models/appeal.rb]: Thinking about this a little more, I think i'd pull the whole ramp election:

```
# RAMP election sent out to potentially turn this appeal into a RAMP HLR or SC
def pending_ramp_election
  RampElection.incomplete.where(veteran_file_number: sanitized_vbms_id).first
end
```

The logic all makes more sense inside of `ramp_election.rb`. That's what it's there for:
```
class RampElection
  # We softly recommend veterans return their RAMP Election within 60 days of the notice date
  SOFT_RETURN_BY_TIME = 60.days

  def soft_due_date
    notice_date + 60.days
  end
end
```

**Comment by cmgiven** [comment]
@shanear: Back to you.

File: app/models/ramp_election.rb [hunk 3]
```diff
@@ -45,6 +47,12 @@ def end_product_description
     end_product_reference_id && end_product.description_with_routing
   end
 
+  # RAMP letters request that Veterans respond within 60 days; elections will
+  # be accepted after this point, however, so this "due date" is soft.
```
Comment (shanear) [inline app/models/ramp_election.rb]: 💯

**Pull request merged**